### ISSUE TITLE
Ubuntu 20.04 iaito package SIGSEGV

### ISSUE THREAD

**Issue by ebubekirtrkr** [radareorg/iaito#61]
#### iaito -v
```sh
Warning: QT_DEVICE_PIXEL_RATIO is deprecated. Instead use:
   QT_AUTO_SCREEN_SCALE_FACTOR to enable platform plugin controlled per-screen factors.
   QT_SCREEN_SCALE_FACTORS to set per-screen factors.
   QT_SCALE_FACTOR to set the application global scale factor.
iaito 5.2.2
```
#### date
```sh
Tue 16 Nov 2021 11:51:14 +03
```
#### r2 -v
```sh
radare2 5.5.0 27096 @ linux-x86-64 git.5.5.0
commit: 645835794b0013b52042d5762d98d03fcbf250f6 build: 2021-11-16__11:08:25
```
#### uname -ms
```sh
Linux x86_64
```
#### lsb_release -d   
```sh
Description:    Ubuntu 20.04.3 LTS
```

## Description

I've installed  r2 using 
```sh
./sys/install.sh  
```
iaito using deb package 
[5.3.1/iaito-amd64.deb.zip](https://github.com/radareorg/iaito/releases/download/5.3.1/iaito-amd64.deb.zip)  

But when running it it complain about `libr_core.so` so I added ` /usr/local/lib` to `/etc/ld.so.conf` , then run `ldconfig` with refer to [iaito/issues/59](https://github.com/radareorg/iaito/issues/59) 

now running iaito got  SIGSEGV
#### iaito   
```sh
Warning: QT_DEVICE_PIXEL_RATIO is deprecated. Instead use:
   QT_AUTO_SCREEN_SCALE_FACTOR to enable platform plugin controlled per-screen factors.
   QT_SCREEN_SCALE_FACTORS to set per-screen factors.
   QT_SCALE_FACTOR to set the application global scale factor.
fish: Job 1, 'iaito' terminated by signal SIGSEGV (Address boundary error)
```

Also I've tried build iaito from github, but it gives me following error
```sh
g++ -c -pipe -O2 -Wall -W -D_REENTRANT -fPIC -DIAITO_SOURCE_BUILD -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -DQT_NO_DEBUG -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I../src -I. -I../src -I../src/core -I../src/widgets -I../src/dialogs -I../src/common -I../src/plugins -I../src/menus -isystem /usr/local/include/libr -isystem /usr/local/include/libr/sdb -isystem /usr/include/libr -isystem /usr/include/libr/sdb -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtSvg -isystem /usr/include/x86_64-linux-gnu/qt5/QtWidgets -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtNetwork -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -I. -I. -I/usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -o MainWindow.o ../src/core/MainWindow.cpp
In file included from /usr/local/include/libr/r_util.h:33,
                 from /usr/local/include/libr/r_getopt.h:8,
                 from /usr/local/include/libr/r_main.h:7,
                 from /usr/local/include/libr/r_core.h:6,
                 from ../src/core/IaitoCommon.h:8,
                 from ../src/common/TempConfig.h:5,
                 from ../src/core/Iaito.cpp:13:
../src/core/Iaito.cpp: In member function ‘QList<RelocDescription> IaitoCore::getAllRelocs()’:
../src/core/Iaito.cpp:2746:27: error: cannot convert ‘r_crbtree_t*’ to ‘RBNode*’ {aka ‘r_rb_node_t*’}
 2746 |         r_rbtree_foreach (relocs, iter, br, RBinReloc, vrb) {
      |                           ^~~~~~
      |                           |
      |                           r_crbtree_t*
/usr/local/include/libr/r_util/r_rbtree.h:63:37: note:   initializing argument 1 of ‘RBIter r_rbtree_first(RBNode*)’
   63 | R_API RBIter r_rbtree_first(RBNode *root);
      |                             ~~~~~~~~^~~~
In file included from /usr/local/include/libr/r_main.h:6,
                 from /usr/local/include/libr/r_core.h:6,
                 from ../src/core/IaitoCommon.h:8,
                 from ../src/common/TempConfig.h:5,
                 from ../src/core/Iaito.cpp:13:
../src/core/Iaito.cpp:2746:56: error: ‘RBinReloc’ {aka ‘struct r_bin_reloc_t’} has no member named ‘vrb’
 2746 |         r_rbtree_foreach (relocs, iter, br, RBinReloc, vrb) {
      |                                                        ^~~
../src/core/Iaito.cpp:2746:56: error: ‘RBinReloc’ {aka ‘struct r_bin_reloc_t’} has no member named ‘vrb’
../src/core/Iaito.cpp:2746:9: error: expected primary-expression before ‘__typeof__’
 2746 |         r_rbtree_foreach (relocs, iter, br, RBinReloc, vrb) {
      |         ^~~~~~~~~~~~~~~~
../src/core/Iaito.cpp:2746:9: error: expected ‘)’ before ‘__typeof__’
In file included from /usr/include/dlfcn.h:24,
                 from /usr/local/include/libr/r_lib.h:8,
                 from /usr/local/include/libr/r_core.h:13,
                 from ../src/core/IaitoCommon.h:8,
                 from ../src/common/TempConfig.h:5,
                 from ../src/core/Iaito.cpp:13:
../src/core/Iaito.cpp:2746:56: error: ‘RBinReloc’ {aka ‘struct r_bin_reloc_t’} has no member named ‘vrb’
 2746 |         r_rbtree_foreach (relocs, iter, br, RBinReloc, vrb) {
      |                                                        ^~~
g++ -c -pipe -O2 -Wall -W -D_REENTRANT -fPIC -DIAITO_SOURCE_BUILD -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -DQT_NO_DEBUG -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I../src -I. -I../src -I../src/core -I../src/widgets -I../src/dialogs -I../src/common -I../src/plugins -I../src/menus -isystem /usr/local/include/libr -isystem /usr/local/include/libr/sdb -isystem /usr/include/libr -isystem /usr/include/libr/sdb -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtSvg -isystem /usr/include/x86_64-linux-gnu/qt5/QtWidgets -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtNetwork -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -I. -I. -I/usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -o Helpers.o ../src/common/Helpers.cpp
make[1]: *** [Makefile:3521: Iaito.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/home/USERNAME/.local/share/radare2/r2pm/git/iaito/build'
make: *** [Makefile:53: iaito] Error 2

```

**Comment by trufae** [comment]
Thats expected, the binary package was built for r2 5.3.x, every minor version change there are abi breakups, so both need to be the same version.

I'll add ifdefs to fix the iaito build, thanks for reporting, the breaking commit was introduced after the 5.5.0 release, so you can rollback a couple of commits in your r2 build or wait until i find time to do a new build of iaito ( i have a list of things to fix/implement in here).

thanks for reporting!

**Comment by ebubekirtrkr** [comment]
Okay, thank you for answering

**Comment by trufae** [comment]
Fixed in c368635b9513d9c51d7e894e8a00ec13da1f81c5
